### PR TITLE
Rewrite some unit tests to criterion based

### PR DESCRIPTION
**Pull request by Kokan**
This PR has a few changes that is not a criterion conversion. 
* Tweaks the Travis.
* "Performance" tests take too long, while does not catch perf issues at all, so the iteration is decreased 
* The control command tests should use only the interface and shall not include `. c` files either.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
Performance test: Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: lib/rewrite/tests/test_rewrite.c [hunk 1]
```diff
@@ -97,277 +95,263 @@ rewrite_teardown(LogMessage *msg)
...
+  do { \
+   gssize value_len = -1; \
+   const gchar *actual = log_msg_get_value_by_name(msg, name, &value_len); \
+   \
+   gssize exp_value_len = expected_value_len; \
+   if (exp_value_len < 0) \
+     exp_value_len = strlen(expected); \
+   \
+   cr_assert_eq(value_len, exp_value_len, "Message name(%s) value_len(%ld) expected_value_len(%ld) at line %d", name, value_len, exp_value_len, __LINE__); \
+   cr_assert_str_eq(actual, expected, "Message name(%s) actual(%s) expected(%s) at line %d", name, actual, expected, __LINE__); \
```
Comment (MrAnno) [inline lib/rewrite/tests/test_rewrite.c]: Is `__LINE__` required here? Without it, this macro can be refactored into an assert function.

File: lib/control/tests/test_control_cmds.c [hunk 2]
```diff
@@ -22,46 +22,62 @@
...
 test_log(void)
 {
   GString *command = g_string_sized_new(128);
   GString *reply;
 
+  CommandFunction control_connection_message_log = command_test_get("LOG")->func;
+
   g_string_assign(command,"LOG");
-  reply = control_connection_message_log(command, NULL);
+  reply = (*control_connection_message_log)(command, NULL);
```
Comment (MrAnno) [inline lib/control/tests/test_control_cmds.c]: I'd remove all explicit function dereference operators from this test, it's redundant.

File: lib/rewrite/tests/test_rewrite.c [hunk 3]
```diff
@@ -97,277 +95,263 @@ rewrite_teardown(LogMessage *msg)
...
-  test_set_field_exist_and_group_set_multiple_fields_with_glob_question_mark_pattern_literal_string();
-  test_set_field_exist_and_group_set_multiple_fields_with_multiple_glob_pattern_literal_string();
-  test_set_field_exist_and_group_set_template_string();
-  test_set_field_exist_and_group_set_template_string_with_old_value();
-  test_set_field_invalid_template();
-  test_set_field_exist_and_group_set_when_condition_doesnt_match();
-  test_set_field_exist_and_group_set_when_condition_matches();
-  test_set_field_cloned();
-  test_unset_field_disappears();
-  test_groupunset_field_disappears();
```
Comment (MrAnno) [inline lib/rewrite/tests/test_rewrite.c]: Nice :)

File: lib/control/tests/test_control_cmds.c [hunk 4]
```diff
@@ -22,79 +22,114 @@
...
+  msg_init(FALSE);
   GString *command = g_string_sized_new(128);
   GString *reply;
 
+  CommandFunction control_connection_message_log = command_test_get("LOG")->func;
+
   g_string_assign(command,"LOG");
-  reply = control_connection_message_log(command, NULL);
-  assert_string(reply->str, "Invalid arguments received, expected at least one argument", "Bad reply");
+  reply = (*control_connection_message_log)(command, NULL);
```
Comment (MrAnno) [inline lib/control/tests/test_control_cmds.c]: I'd remove all explicit function dereference operators from this test, it's redundant.

File: lib/scanner/kv-scanner/tests/test_kv_scanner.c [hunk 5]
```diff
@@ -966,7 +966,7 @@ _provide_cases_for_performance_test_parse_long_msg(void)
   return g_memdup(tc, sizeof(tc));
 }
 
-#define ITERATION_NUMBER 100000
+#define ITERATION_NUMBER 10000
```
Comment (furiel) [inline lib/scanner/kv-scanner/tests/test_kv_scanner.c]: As discussed irl, please drop this patch.

File: lib/rewrite/tests/test_rewrite.c (hunk 1 above)
Comment (MrAnno) [reply lib/rewrite/tests/test_rewrite.c]: Without `__LINE__`, refactored into a separate function, it looks like something like this:

```
[----] ../lib/rewrite/tests/test_rewrite.c:123: Assertion failed: Message device_id(00100) value_len(0) -1(5)
[FAIL] rewrite::condition_success: (0,01s)
```

The line number here points to the assert function, not to its caller, but the call side can be found easily based on the actual parameters.

What do you think?

Comment (Kokan) [reply lib/rewrite/tests/test_rewrite.c]: That's fine by me. Before criterion it was painful.

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
success

**Pull request merged**